docs/coding-standards/js.md

```diff
@@ -1,61 +1,164 @@
 
-### Visual Studio Code
+```js
+// bad
+Checkboxes.prototype = {
+  init: function () {
+    ...
+  }
+}
 
-Install [vscode-standardjs](https://marketplace.visualstudio.com/items/chenxsan.vscode-standardjs). It includes support for automatic formatting.
+// good
+Checkboxes.prototype.init = function () {
+  ...
+}
+```
 
-For snippets, install [vscode-standardjs-snippets](https://marketplace.visualstudio.com/items?itemName=capaj.vscode-standardjs-snippets).
+When initialising an object, use the `new` keyword.
 
-### Other editors
+```js
+// bad
+var myCheckbox = Checkbox().init()
 
-There are [official guides for most of the popular editors](http://standardjs.com/index.html#text-editor-plugins).
+// good
+var myCheckbox = new Checkbox().init()
+```
+## Modules
 
-## Do I need to respect this?
+Use ES6 modules (`import`/`export`) over a non-standard module system. You can always transpile to your preferred module system.
 
-If you want to submit a pull request to the govuk-frontend, your code will need to pass the linter.
+```js
+import { nodeListForEach } from '../globals/common'
+...
+export default Checkboxes
+```
 
-If you're just using the prototype kit in a separate project, then no, you aren't forced to use standard, or any other linter for that matter. Just write code as you would normally.
+Avoid using wildcard (`import * as nodeListForEach`) imports.
 
-## Why lint?
+Use default export over named export.
 
-Automated linting ensures project-wide consistency and limits (ideally eliminates) bikeshedding discussions involving spacing, naming conventions, quotes, and others during the pull request review process. It frees the reviewer to focus on the actual substance rather than stylistic issues.
+## Polyfilling
 
-More importantly, linting will catch some low hanging programmer errors, such as calling an undefined function or assigning a value and then never reading it. These allow the programmer to catch some bugs before having to test the code.
+If you’re new to polyfilling, start by reading [this explanation](https://remysharp.com/2010/10/08/what-is-a-polyfill).
 
-## Why standard?
+Before GOV.UK Frontend, our projects used jQuery for: DOM interactions, events and data manipulation.
 
-Linting rules can be a contentious subject, and a lot of them are down to personal preference. The core idea of standard is to be opinionated and limit the amount of initial bikeshedding discussions around which linting rules to pick, because in the end, it's not as important which rules you pick as it is to just be consistent about it. This is why we chose standard: because we want to be consistent about how we write code, but don't want to spend unnecessary time picking different rules (which all have valid points).
+We’re taking a step back from jQuery due to its lack of support for the browsers we support, its large file size, lack of security updates and from conversations with the community.
```

Comment:
lack of security updates is just 1.x no? It's not a general jQuery problem is it?

Reply:
Is it useful to talk about our previous approach so much? Why not just go straight into the new thing?

Reply:
Lack o security updates is 2.x too, which is the one needed for IE8 support.
I guess it's making a case about why we decided to go that way.

Reply:
IE8 support requires 1.x.

Reply:
yep my point is just that it currently reads that jQuery has a 'lack of security updates' which makes it sound bad. jQuery has security updates but not in the old version we need for IE support.